revocation/internal/ocsp/ocsp.go

```diff
@@ -0,0 +1,234 @@
+	// Max size determined from https://www.ibm.com/docs/en/sva/9.0.6?topic=stanza-ocsp-max-response-size.
+	// Typical size is ~4 KB
+	ocspMaxResponseSize int64 = 20480 //bytes
+)
+
+// CertCheckStatus checks the revocation status of a certificate using OCSP
+func CertCheckStatus(cert, issuer *x509.Certificate, opts CertCheckStatusOptions) *result.CertRevocationResult {
+	if !Supported(cert) {
+		// OCSP not enabled for this certificate.
+		return &result.CertRevocationResult{
+			Result:        result.ResultNonRevokable,
+			ServerResults: []*result.ServerResult{toServerResult("", NoServerError{})},
+		}
+	}
+	ocspURLs := cert.OCSPServer
+
+	serverResults := make([]*result.ServerResult, len(ocspURLs))
+	for serverIndex, server := range ocspURLs {
+		serverResult := checkStatusFromServer(cert, issuer, server, opts)
+		if serverResult.Result == result.ResultOK ||
+			serverResult.Result == result.ResultRevoked ||
+			(serverResult.Result == result.ResultUnknown && errors.Is(serverResult.Error, UnknownStatusError{})) {
+			// A valid response has been received from an OCSP server
+			// Result should be based on only this response, not any errors from
+			// other servers
+			return serverResultsToCertRevocationResult([]*result.ServerResult{serverResult})
+		}
+		serverResults[serverIndex] = serverResult
+	}
+	return serverResultsToCertRevocationResult(serverResults)
+}
+
+// Supported returns true if the certificate supports OCSP.
+func Supported(cert *x509.Certificate) bool {
+	return cert != nil && len(cert.OCSPServer) > 0
+}
+
+func checkStatusFromServer(cert, issuer *x509.Certificate, server string, opts CertCheckStatusOptions) *result.ServerResult {
+	// Check valid server
+	if serverURL, err := url.Parse(server); err != nil || !strings.EqualFold(serverURL.Scheme, "http") {
+		// This function is only able to check servers that are accessible via HTTP
+		return toServerResult(server, GenericError{Err: fmt.Errorf("OCSPServer protocol %s is not supported", serverURL.Scheme)})
+	}
+
+	// Create OCSP Request
+	resp, err := executeOCSPCheck(cert, issuer, server, opts)
+	if err != nil {
+		// If there is a server error, attempt all servers before determining what to return
+		// to the user
+		return toServerResult(server, err)
+	}
+
+	// Validate OCSP response isn't expired
+	if time.Now().After(resp.NextUpdate) {
+		return toServerResult(server, GenericError{Err: errors.New("expired OCSP response")})
+	}
+
+	// Handle pkix-ocsp-no-check and id-ce-invalidityDate extensions if present
+	// in response
+	extensionMap := extensionsToMap(resp.Extensions)
+	if _, foundNoCheck := extensionMap[pkixNoCheckOID]; !foundNoCheck {
+		// This will be ignored until CRL is implemented
+		// If it isn't found, CRL should be used to verify the OCSP response
+		_ = foundNoCheck // needed to bypass linter warnings (Remove after adding CRL)
+		// TODO: add CRL support
+		// https://github.com/notaryproject/notation-core-go/issues/125
+	}
```

Comment:
Should this part be properly handled by this PR?

Reply:
Could you open an issue and address it in another PR?

Reply:
Created the issue #220